tuf/formats.py

```diff
@@ -684,161 +673,41 @@ def make_metadata(version, expiration_date, filedict):
-    SNAPSHOT_SCHEMA.check_match(object)
-    
-    version = object['version']
-    expires = object['expires']
-    versiondict = object['meta']
-    
-    return SnapshotFile(version, expires, versiondict)
-
-
-  @staticmethod
-  def make_metadata(version, expiration_date, versiondict):
-    result = {'_type' : 'Snapshot'}
-    result['version'] = version 
-    result['expires'] = expiration_date
-    result['meta'] = versiondict
-
-    # Is 'result' a Snapshot metadata file?
-    # Raise 'tuf.FormatError' if not.
-    SNAPSHOT_SCHEMA.check_match(result)
-    
-    return result
-
-
-
-
-
-class TargetsFile(MetaFile):
-  def __init__(self, version, expires, filedict=None, delegations=None):
-    if filedict is None:
-      filedict = {}
-    if delegations is None:
-      delegations = {}
-    self.info = {}
-    self.info['version'] = version
-    self.info['expires'] = expires
-    self.info['targets'] = filedict
-    self.info['delegations'] = delegations
-
-
-  @staticmethod
-  def from_metadata(object):
-    # Is 'object' a Targets metadata file?
-    # Raise tuf.FormatError if not.
-    TARGETS_SCHEMA.check_match(object)
-    
-    version = object['version']
-    expires = object['expires']
-    filedict = object.get('targets')
-    delegations = object.get('delegations')
-    
-    return TargetsFile(version, expires, filedict, delegations)
-
-
-  @staticmethod
-  def make_metadata(version, expiration_date, filedict=None, delegations=None):
-    if filedict is None and delegations is None:
-      raise tuf.Error('We don\'t allow completely empty targets metadata.')
-
-    result = {'_type' : 'Targets'}
-    result['version'] = version
-    result['expires'] = expiration_date
-    result['targets'] = {} 
-    if filedict is not None:
-      result['targets'] = filedict
-    if delegations is not None:
-      result['delegations'] = delegations
-
-    # Is 'result' a Targets metadata file?
-    # Raise 'tuf.FormatError' if not.
-    TARGETS_SCHEMA.check_match(result)
-    
-    return result
-
-
+  Checks the result to make sure that it conforms to the given schema, raising
+  an error if not.
 
+  Returns the new dict conforming to the schema if there are no problems.
+  """
 
+  # Check that schema supports a check_match call.
+  # Duck typing version of this check:
+  if not hasattr(schema, 'check_match'):
+    raise ValueError(
+        'The given "schema" does not seem to be a schema.  It has no '
+        '"check_match" method.  Given schema: ' + repr(schema))
 
-class MirrorsFile(MetaFile):
-  def __init__(self, version, expires):
-    self.info = {}
-    self.info['version'] = version
-    self.info['expires'] = expires
+  # # Strict typing version of this check:
+  # # Check that schema_name is a SCHEMA.Object.
+  # if not isinstance(schema, schema.Schema):
+  #   raise ValueError(
+  #       'The first argument must be a schema.Schema object, but is not. '
+  #       'Given schema: ' + repr(schema))
```

Comment:
Just trying to be a good Python boy, I guess. /:  All I really require is that whatever the object is supports check_match, and so that is all I'm asking for.  I included the alternative code because it's a principle I'm less than sure about. :)  What do you think?

Reply:
AFAIK the good Python boy *"doesn't ask for permission but for forgiveness"*:

```python
try:
  schema.check_match(kwargs)

except AttributeError:
  raise ValueError("...")
```